Automattic/woocommerce-payments
# Using onClick instead of action so tasks works on non-expandable mode

Fixes #4561

#### Changes proposed in this Pull Request

This PR includes no change in the UI, it only fixes "Update WooCommerce Payments business details" not working in WooCommerce Home page.
I updated the tasks list logic to use `onClick` when an `action` is not provided, similar to what WooCommerce does.
With that, I renamed the `action` property on custom tasks to `onClick`. This will make they behave the same in expandable and non-expandable modes (with the button being the only difference).

#### Testing instructions

1. Make your account require update so wcpay display the message. This can be faked by following these steps:
    1. Remove conditional check for the task to appear [here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/overview/task-list/tasks.js#L53-L54).
    1. Set the action as not completed by updating the code [here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/overview/task-list/tasks.js#L62) and [here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/overview/task-list/tasks.js#L64).
    1. It should look like this:
        ```js
        return [
            {
                key: 'update-business-details',
                ...
                completed: false,
                onClick:
                    false
                        ? undefined
                        : () => {
                                window.open( accountLink, '_blank' );
                          },
        ```
1. Go to http://localhost:8082/wp-admin/admin.php?page=wc-admin
1. Click "Update WooCommerce Payments business details" and make sure it redirects you to Stripe. (this was not working before)
1. Go to http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview
1. Click "Update" and make sure it still redirects you to Stripe. (this was working already, just make sure it is not broken)

-------------------


**Post merge**
